### PR TITLE
Added MDK4 for Termux

### PR DESCRIPTION
**Pull request by gitdebpro**
added a new package called MDK4 tested on armhf but not on arm64 due to lack of device availability.

### CODE REVIEW

**Comment by gitdebpro** [comment]
Hi there attaching mdk4 build script with all patches and also already build mdk4.deb for armhf only. Just extract the zip.
[mdk4.zip](https://github.com/termux/termux-root-packages/files/6312659/mdk4.zip)

File: packages/mdk4/build.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#MDK is a proof-of-concept tool to exploit common IEEE 802.11 protocol weaknesses.
```
Comment (Grimler91) [inline packages/mdk4/build.sh:1]: Remove all of these comments

File: packages/mdk4/build.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# MDK4 is a new version of MDK3.
+
+#MDK4 is a Wi-Fi testing tool from E7mer of 360, ASPj of k2wrlz, it uses the osdep library from the aircrack-ng project to inject frames on several operating systems. Many parts of it have been contributed by the great aircrack-ng community: Antragon, moongray, Ace, Zero_Chaos, Hirte, thefkboss, ducttape, telek0miker, Le_Vert, sorbo, Andy Green, bahathir, Dawid Gajownik and Ruslan Nabioullin. THANK YOU!
+
+#MDK4 is licenced under the GPLv2 or later.
+
+TERMUX_PKG_MAINTAINER="KimoCoder & Ezmer "
```
Comment (Grimler91) [inline packages/mdk4/build.sh:8]: Are you Kimocoder or Ezmer? 

If not then set it you yourself or @termux. It should be the person responsible for the package in the termux repositories.

File: packages/mdk4/build.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+TERMUX_PKG_HOMEPAGE=https://github.com/aircrack-ng/mdk4
+TERMUX_PKG_DESCRIPTION="MDK is a proof-of-concept tool to exploit common IEEE 802.11 protocol weaknesses."
+TERMUX_PKG_LICENSE="GPL-3.0"
+
+
```
Comment (Grimler91) [inline packages/mdk4/build.sh:13]: Remove extra empty lines

File: packages/mdk4/build.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+TERMUX_PKG_VERSION=4.1
+TERMUX_PKG_REVISION=1
```
Comment (Grimler91) [inline packages/mdk4/build.sh:15]: Remove this line, it does not have to be set for new packages

File: packages/mdk4/build.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+TERMUX_PKG_SRCURL=https://github.com/aircrack-ng/mdk4/archive/master.zip
```
Comment (Grimler91) [inline packages/mdk4/build.sh:16]: Don't use master.zip. Build from a tagged release or a specific commit

File: packages/mdk4/build.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+TERMUX_PKG_SHA256=48708cbf07b324b689b6449ebea10ff2bba5f41e474d63583511812fe5829290
+TERMUX_PKG_DEPENDS="aircrack-ng, pkg-config, libnl, libpcap"
```
Comment (Grimler91) [inline packages/mdk4/build.sh:18]: Remove pkg-config. It is not a runtime dependency

File: packages/mdk4/build.sh [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+TERMUX_PKG_BUILD_IN_SRC=true
```
Comment (Grimler91) [inline packages/mdk4/build.sh:19]: Have you tried building out of source, i.e. without this line?

File: packages/mdk4/build.sh [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
```
Comment (Grimler91) [inline packages/mdk4/build.sh:20]: Remove the extra empty line here

**Review by Grimler91 — CHANGES_REQUESTED** [review]
See comments

File: packages/mdk4/build.sh (hunk 1 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh:1]: ok will do that

File: packages/mdk4/build.sh (hunk 7 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh:19]: no i didn't try that

Comment (Grimler91) [reply packages/mdk4/build.sh:19]: Please try it. This option should only be set if it is required

File: packages/mdk4/build.sh (hunk 1 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh:1]: removed

File: packages/mdk4/build.sh (hunk 2 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh:8]: used @termux

File: packages/mdk4/build.sh (hunk 3 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh:13]: removed

File: packages/mdk4/build.sh (hunk 4 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh:15]: without these lines there are build error so kept these lines but turned the value empty

Comment (Grimler91) [reply packages/mdk4/build.sh:15]: TERMUX_PKG_VERSION=4.1 needs to be there. TERMUX_PKG_REVISION=1 can be removed

Comment (gitdebpro) [reply packages/mdk4/build.sh:15]: ok got it will do that

Comment (gitdebpro) [reply packages/mdk4/build.sh]: removing the line TERMUX_PKG_REVISION=1 stops the build process but keeping its value empty dosent affect the build process so I kept it empty and also tested arm and aarch64 both developed correctly

File: packages/mdk4/build.sh (hunk 5 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh]: changed to commit fc03b0ece893fd8d3f546d915511f01ba3faebc8.zip

File: packages/mdk4/build.sh (hunk 6 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh]: removed

File: packages/mdk4/build.sh (hunk 7 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh:10]: tried without these lines but error in build process

File: packages/mdk4/build.sh (hunk 8 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh]: removed

**Comment by gitdebpro** [comment]
Finally completed all changes as requested and tested on both arm and aarch64 , both completed and created respective .deb files successfully

**Comment by gitdebpro** [comment]
@Grimler91 I just want to know if accepted, when this package will be available at termux package repo i mean when it will show inside termux like 

apt-cache search mdk4 will show it.

Regards.

File: packages/mdk4/build.sh [hunk 9]
```diff
@@ -0,0 +1,10 @@
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_HOMEPAGE=https://github.com/aircrack-ng/mdk4
+TERMUX_PKG_DESCRIPTION="MDK is a proof-of-concept tool to exploit common IEEE 802.11 protocol weaknesses."
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_VERSION=4.1
+TERMUX_PKG_REVISION=
```
Comment (Grimler91) [inline packages/mdk4/build.sh:6]: Remove this line completely, no need to set it

**Review by Grimler91 — CHANGES_REQUESTED** [review]

**Comment by Grimler91** [comment]
@gitdebpro it should be available on all mirrors within a few hours after adding it

File: packages/mdk4/build.sh (hunk 9 above)
Comment (gitdebpro) [reply packages/mdk4/build.sh]: removed done.

**Comment by gitdebpro** [comment]
I have made all the suggested changes now I guess there is no problem.

Regards

**Comment by Grimler91** [comment]
Great, thanks for the PR!

**Pull request merged**